src/app/profile/overview/work-items/work-items.component.ts

```diff
@@ -84,28 +85,43 @@ export class WorkItemsComponent implements OnDestroy, OnInit {
   }
 
   private fetchWorkItemsBySpace(space: Space): void {
-    let filters = [
-      {
-        paramKey: 'filter[assignee]',
-        value: this.context.user.id,
-        active: true
-      }
-    ];
-    this.workItemService._currentSpace = space;
+
+    const assigneeQuery = this.filterService.queryJoiner(
+      {},
+      this.filterService.and_notation,
+      this.filterService.queryBuilder(
+        'assignee', this.filterService.equal_notation, this.context.user.id
+      )
+    );
+    const spaceQuery = this.filterService.queryBuilder(
+      'space', this.filterService.equal_notation, space.id
+    );
+    const filters = this.filterService.queryJoiner(
+      assigneeQuery, this.filterService.and_notation, spaceQuery
+    );
     this.subscriptions.push(
       this.workItemService
-        .getWorkItems(100000, filters)
-        .subscribe(
-          (result: {
-            workItems: WorkItem[];
-            nextLink: string;
-            totalCount?: number;
-            included?: WorkItem[];
-            ancestorIDs?: string[];
-          }): void => {
-            this.workItems = filterOutClosedItems(result.workItems);
-          }
-        )
+        .getWorkItems(100000, {expression: filters})
+        .map(val => val.workItems)
+        .map(workItems => filterOutClosedItems(workItems))
+        // Resolve the work item type, creator and area
```

Comment:
This seems like a potential performance issue wherein for each work item you're now fetching from the server all this extra information.

Why is this info now necessary when previously it wasn't?

Reply:
Good catch. We don't use resolvers in planner anymore. Thus I moved all those operations for resolving in here. But true, that we are resolving more than we need. I'll fix that.